example_school_library_decorator/app.rb

```diff
@@ -0,0 +1,118 @@
+require_relative './book'
+require_relative './person'
+require_relative './student'
+require_relative './teacher'
+require_relative './rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  # List all books
+  def list_all_books
+    if @books.length.positive?
+      @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+
+    else
+      puts 'There are no books at the moment'
+    end
+  end
+
+  # List all people
+  def list_all_people
+    if @people.length.positive?
+      @people.each do |person|
+        puts "name: #{person.name}, age: #{person.age}, id: #{person.id}"
+      end
+    else
+      puts 'Sorry! cannot find person'
+    end
+  end
```

Comment:
### **Highlight**
Well done refactoring the logic out and following the Single responsibility rule. 💯